compiler/z/codegen/OpMemToMem.cpp

```diff
@@ -1517,18 +1527,19 @@ MemCmpVarLenMacroOp::generate(TR::Register* dstReg, TR::Register* srcReg, TR::Re
    TR::Compilation *comp = _cg->comp();
 
    if(cursorBefore == NULL) cursorBefore = _cg->getAppendInstruction();
+   generateShortCircuitInstructions(_cg, _rootNode, _resultReg, _srcReg, _dstReg, _doneLabel);
    generateLoop();
    setInRemainder(true);
    generateRemainder();
 
    _cursor = generateCmpResult(_cg, _rootNode, _dstReg, _srcReg, _resultReg, _falseLabel, _trueLabel, _doneLabel);
    TR::RegisterDependencyConditions * dependencies = generateDependencies();
    _cursor->setDependencyConditions(dependencies);
-   if(_startControlFlow==NULL)
```

Comment:
The if is not necessary any more since the control flow always start from the short circuit instructions